packages/block-editor/src/components/provider/index.js

```diff
@@ -34,10 +34,10 @@ class BlockEditorProvider extends Component {
 			this.attachChangeObserver( registry );
 		}
 
-		if ( this.isSyncingOutcomingValue ) {
-			this.isSyncingOutcomingValue = false;
+		if ( this.isSyncingOutcomingValue === value ) {
```

Comment:
> Because we don't update `this.isSyncingOutcomingValue` when the new value is different, an update where the value is set to the last outcoming value or `null` would be ignored. Certain implementations of "undo" could be affected by this (maybe ours?).

That's a good point. Should only the addition of `this.isSyncingOutcomingValue = null;` to the `else if` condition be sufficient though? Is testing for `!== null` necessary in the first condition?

Reply:
I wonder if we could just form it like this, where assigning to `null` is fine to _always_ occur in `componentDidUpdate`, after the logic to consider whether the value has changed, is not a result of an outgoing sync.

```js
componentDidUpdate( prevProps ) {
	// ...

	if ( value !== prevProps.value && this.isSyncingOutcomingValue !== value ) {
		this.isSyncingIncomingValue = value;
		resetBlocks( value );
	}

	this.isSyncingOutcomingValue = null;
}
```

Reply:
>Should only the addition of this.isSyncingOutcomingValue = null; to the else if condition be sufficient though? 

I think so.

>Is testing for !== null necessary in the first condition?

Yeah, otherwise, the parent setting blocks to `null` would be ignored if there is no pending outcoming value.

>I wonder if we could just form it like this, where assigning to null is fine to always occur in componentDidUpdate, after the logic to consider whether the value has changed, is not a result of an outgoing sync.

This will force a double reset if there is any update scheduled by React in-between when an outcoming sync starts and ends.

Reply:
> This will force a double reset if there is any update scheduled by React in-between when an outcoming sync starts and ends.

That's fair. I'm not sure we expect that to happen, but it's probably worth the precaution. I'll revise accordingly.

Reply:
Yeah, before, you could get a double reset either from the parent modifying the value or from there being an update in-between that cleared `isSyncingOutcomingValue`.

React updates can get pretty crazy, even if they can be deterministic at times, the order is so hard to wrap your head around that we are better off just being defensive.